libs/ml/tests/ml/layers/PRelu.cpp

```diff
@@ -230,14 +230,16 @@ TYPED_TEST(PReluTest, saveparams_test)
   TypeParam loss = layer.Evaluate(error_output);
   layer.BackPropagate(error_output);
   layer.ApplyRegularisation();
-  layer.Step(DataType{0.1f});
+  auto grads = layer.GetGradients();
+  layer.ApplyGradients(grads);
 
   // train g2
   layer2.SetInput(label_name, labels);
   TypeParam loss2 = layer2.Evaluate(error_output);
   layer2.BackPropagate(error_output);
   layer2.ApplyRegularisation();
-  layer2.Step(DataType{0.1f});
+  auto grads2 = layer2.GetGradients();
+  layer2.ApplyGradients(grads2);
```

Comment:
I've just realised that you need to multiply gradient by -LEARNING_RATE otherwise you are maximising the error.

Reply:
Or use Optimiser which will do it for you.

Reply:
oh that's true. it actually makes no difference for this test; but it's good to demonstrate correct usage :-p I'll fix that